Sources/SwiftDocC/Infrastructure/DocumentationContext.swift

```diff
@@ -2535,6 +2545,9 @@ public class DocumentationContext: DocumentationContextDataProviderDelegate {
             let referenceWithoutFragment = reference.withFragment(nil)
             return try entity(with: referenceWithoutFragment).availableSourceLanguages
         } catch ContextError.notFound {
+            if let externalEntity = externalCache[reference.withFragment(nil)] {
```

Comment:
Thanks for pointing this out. The more I look at this the more I think that this code shouldn't use `withFragment(nil)` because the external resolver adds both pages and on-page elements to the same cache.